### ISSUE TITLE
visualize sample from test set

### ISSUE THREAD

**Issue by taeil** [Berkeley-Data/irrigation_detection#43]
- use wandb image logging 
- pick top/bottom examples for fp, fn, tp, tn

**Comment by suryatechie** [comment]
Today, I tried to analyze the input TFRecord dataset. The idea is to extract the image and current label information to validate against the prediction. Based on that, we can extract some false positive and false negative examples and push them to wandb. The tensors are having 128 shapes. The following is the binary label tensor for an example:
binary_label tf.Tensor(
[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0], shape=(128,), dtype=int64)

**Comment by taeil** [comment]
fyi, [Keras model.evaluate() and model.predict()?](https://stackoverflow.com/questions/44476706/what-is-the-difference-between-keras-model-evaluate-and-model-predict#:~:text=The%20keras.,loss%20value%20for%20every%20batch.&text=predict()%20function%20will%20give,a%20batch%2C%20for%20all%20batches.]

**Comment by suryatechie** [comment]
Tried multiple options. I was able to get individual example data by setting batch size to 1 during model evaluation and using callback functions.  We can take a random sample of examples (once we get fp and fn values, we can stop processing), call model evaluation with batch size set to 1 (indicates 1 example), and collect the individual metrics using callback functions. Sample output from a call back function:
 ['loss', 'tp', 'fp', 'tn', 'fn', 'accuracy', 'precision', 'recall', 'auc']; 
 [0.8092604875564575, 0.0, 1.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]

Next step is to track the image data, process it, push it to wandb if either fp or fn is 1.0

**Comment by taeil** [comment]
how about get whole batch? Batch example is very good to check too.

**Comment by taeil** [comment]
thanks for making progress! This is great!

**Comment by suryatechie** [comment]
That can be done. We can check for one whole batch and see if there are any fp or fn and take some images within that batch. But, we cannot get exactly which image created fp or fn. Since batch size can be fine-tuned, if we really want to nail it down, we can decrease the batch size using the input parameter. I will use the whole batch size and continue the work.

**Comment by suryatechie** [comment]
I ran into technical challenges with batch as I cannot map images within a batch to a batch result as I don't know which batch got processed within a dataset. I will link some sample images from the processed dataset.

**Comment by suryatechie** [comment]
Moving this to Milestone 2.
